src/V3Simulate.h

```diff
@@ -243,8 +243,9 @@ class SimulateVisitor VL_NOT_FINAL : public VNVisitor {
             constp->user2(1);
             freeList.push_back(constp);
         }
-        constp->num().isDouble(nodep->isDouble());
-        constp->num().isString(nodep->isString());
+        constp->num().dataType(nodep->isDouble()   ? V3NumberDataType::DOUBLE
+                               : nodep->isString() ? V3NumberDataType::STRING
+                                                   : V3NumberDataType::LOGIC);
```

Comment:
Perhaps make a standard call that is e.g. num().setDataTypeFromNodep(constp);.  Then can merge that with the above pre-pull, so no non-V3Number code changes with this pull.

Ideally this code also wouldn't be changing the num() format at all, but I don't see obvious alternatives.

Reply:
Fixed in https://github.com/verilator/verilator/pull/3529